src/main/java/org/opentripplanner/model/plan/legreference/LegReferenceSerializer.java

```diff
@@ -51,16 +51,26 @@ public static LegReference decode(String legReference) {
       return null;
     }
 
-    var buf = Base64.getUrlDecoder().decode(legReference);
-    var input = new ByteArrayInputStream(buf);
+    byte[] serializedLegReference;
+    try {
+      serializedLegReference = Base64.getUrlDecoder().decode(legReference);
```

Comment:
Out of interest: are you supposed to create a new decoder for every call or can you re-use the instance? (Doesn't make a difference for the review of this PR)

Reply:
Base64.getUrlDecoder() is a thread-safe singleton instance.